### ISSUE TITLE
How to use EqualsProto to check protobuf?

### ISSUE THREAD

**Issue by yangminzhu** [google/googletest#1761]
The [advanced doc](https://github.com/google/googletest/blame/master/googletest/docs/advanced.md#L407) says `you can use gMock's EqualsProto to compare protos in your tests`, but I find the `EqualsProto` is **not** defined inside gMock but in the nucleus repo that is used for parsing common genomics file.

I don't want to depend on this nucleus repo, so how can I use the `EqualsProto` in my project?

Thanks

**Comment by yangminzhu** [comment]
I don't know if this is supposed to be an internal-only feature, I found it from the public doc: https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#asserting-using-gmock-matchers and the source code is uploaded to https://github.com/google/nucleus/blob/master/nucleus/testing/protocol-buffer-matchers.h.

**Comment by devjgm** [comment]
What's the status of this issue? The [advanced.md file](https://github.com/google/googletest/blob/master/googletest/docs/advanced.md) references using `::testing::EqualsProto`, but this matcher is not part of gmock.

**Comment by neuromage** [comment]
Hi,

Do we have any plans to move and OSS protocol-buffer-matchers.h to googletest?

**Comment by smolkaj** [comment]
Any updates on this? Not having this be open source complicates open sourcing google projects.

**Comment by dayfine** [comment]
Is there a timeline on this (Suppose it's still on the roadmap)? I suppose it might not be as simple as copying over the two files in the necleus repo as I'd imagine the export need to be (somewhat) consistency with the current internal version?

**Comment by jheaff1** [comment]
I too am interested in this

**Comment by mr-salty** [comment]
+1 this would be very useful to the Cloud C++ team.  We have made do so far with our own hand-crafted matchers but having the full suite of proto-matchers would be nicer (and we can delete our own implementations).

**Comment by blais** [comment]
I could have used this too today. Please!

**Comment by zckevin** [comment]
protocol-buffer-matchers.h in Nucleus is heavily integrated with TensorFlow, it's not easy to include it without a rewrite.

**Comment by inazarenko** [comment]
I forked them from Nucleus a while back and removed the unneeded dependencies:

https://github.com/inazarenko/protobuf-matchers

**Comment by yangustc07** [comment]
@inazarenko Are you planning to merge it into the googletest repo? I also have a use case for it.

**Comment by inazarenko** [comment]
> Are you planning to merge it into the googletest repo? I also have a use case for it.

No, I don't plan to merge it with googletest, but you can just pull my repo and use it. Or clone it if you want to be isolated from any changes I might make (although I don't have any current plans to change anything, it works fine as is for our purposes).

**Comment by derekmauro** [comment]
If this is open sourced (and I think it should be), it will be part of the protobuf library, not GoogleTest. There is a lot of cleanup required first though.

**Comment by alexolog** [comment]
At the least, remove this from the documentation to avoid confusing people

**Comment by devjgm** [comment]
Friendly ping. The docs should remove references to EqualsProto() and any other matches that don't exist externally.

**Comment by wffurr** [comment]
Rather than removing the references, can we have the proto matchers as open source?  I would love to use those in my project instead of hacking something together over `MessageDifferencer`.

**Comment by inazarenko** [comment]
I'd also love for them to be a part of gTest/gMock.

@wffurr for now perhaps take a look at the repository linked here: https://github.com/google/googletest/issues/1761#issuecomment-834858070

**Comment by tonyliaoss** [comment]
This seems related to https://github.com/protocolbuffers/protobuf/issues/3142
